core/controllers/reader.py

```diff
@@ -295,27 +300,23 @@ def get(self, exploration_id):
         most_recently_reached_checkpoint_exp_version = None
         most_recently_reached_checkpoint_state_name = None
 
-        # If exp_user_data is None, it means the exploration is started
-        # for the first time and no checkpoint progress of the user exists for
-        # the respective exploration.
-        # Exploration version 'v' passed as a parameter in GET request means a
-        # previous version of the exploration is being fetched. In that case,
-        # we want that exploration to start from the beginning and do not
-        # allow users to save their checkpoint progress in older exploration
-        # version.
-        if exp_user_data is not None and version is None:
+        if not self.user_id and unique_progress_url_id is not None:
```

Comment:
Why are we handling the logged out case in this handler when we are already handling in `SaveTransientCheckpointProgressHandler` ? This handler should not be called in the logged out user case in the first place.

Reply:
Actually, the `SaveTransientCheckpointProgressHandler` is responsible for updating the logged-out progress. When the logged out user is redirected to the Exploration page, we need to populate the exploration page related data in this handler.